routing/router.go

```diff
@@ -1309,6 +1311,37 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 		channelID := lnwire.NewShortChanIDFromInt(msg.ChannelID)
 		fundingTx, err := r.fetchFundingTx(&channelID)
 		if err != nil {
+			// In order to ensure we don't errnosuly mark a channel as
+			// a zmobie due to an RPC failure, we'll attempt to string
+			// match for the relevant errors.
+			//
+			// * btcd:
+			//    * https://github.com/btcsuite/btcd/blob/master/rpcserver.go#L1316
+			//    * https://github.com/btcsuite/btcd/blob/master/rpcserver.go#L1086
+			// * bitcoind:
+			//    * https://github.com/bitcoin/bitcoin/blob/7fcf53f7b4524572d1d0c9a5fdc388e87eb02416/src/rpc/blockchain.cpp#L770
+			//     * https://github.com/bitcoin/bitcoin/blob/7fcf53f7b4524572d1d0c9a5fdc388e87eb02416/src/rpc/blockchain.cpp#L954
+			switch {
+			case strings.Contains(err.Error(), "not found"):
```

Comment:
i'm a little hesitant to introduce string parsing on errors as we've been bitten in the past, but i think that's okay here since it is mostly an optimization 👍

Reply:
Yeah I feel the same, but in this case it's relatively contained. What we _really_ need to do is do all this error code massaging at the `rpcclient` level, so we can have it apply to both `bitcoind` and `btcd`, and the other projects that leverage the library can help maintain the set of error codes.